target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8071-mf269.dts

```diff
@@ -0,0 +1,449 @@
+};
+
+&qpic_nand {
+	status = "okay";
+
+	/*
+	 * Bootloader will find the NAND DT node by the compatible and
+	 * then "fixup" it by adding the partitions from the SMEM table
+	 * using the legacy bindings thus making it impossible for us
+	 * to change the partition table or utilize NVMEM for calibration.
+	 * So add a dummy partitions node that bootloader will populate
+	 * and set it as disabled so the kernel ignores it instead of
+	 * printing warnings due to the broken way bootloader adds the
+	 * partitions.
+	 */
+	partitions {
+		status = "disabled";
+	};
+
+	nand@0 {
+		reg = <0>;
+		nand-ecc-strength = <4>;
+		nand-ecc-step-size = <512>;
+		nand-bus-width = <8>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "fota-flag";
+				reg = <0x0000000 0x00a0000>;
+				read-only;
+			};
+
+			partition@a0000 {
+				label = "mac";
+				reg = <0x00a0000 0x0080000>;
+				read-only;
+
+				nvmem-layout {
+					compatible = "fixed-layout";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					macaddr_mac_0: macaddr@0 {
+						compatible = "mac-base";
+						reg = <0x0 0x6>;
+						#nvmem-cell-cells = <1>;
+					};
+				};
+			};
+
+			partition@120000 {
+				label = "cfg-param";
+				reg = <0x0120000 0x1400000>;
+				read-only;
+			};
+
+			partition@1520000 {
+				label = "log";
+				reg = <0x1520000 0x0600000>;
+				read-only;
+			};
+
+			partition@1b20000 {
+				label = "oops";
+				reg = <0x1b20000 0x00a0000>;
+				read-only;
+			};
+
+			partition@1bc0000 {
+				label = "web";
+				reg = <0x1bc0000 0x0800000>;
+				read-only;
+			};
+
+			partition@23c0000 {
+				label = "ubi_kernel";
+				reg = <0x23c0000 0x3400000>;
+			};
+
+			partition@57c0000 {
+				label = "0:wififw";
+				reg = <0x57c0000 0x0800000>;
+				read-only;
+			};
+
+			/* rootfs partition is the result of squashing
+			 * consecutive stock partitions:
+			 * - openwrt_data (25 MiB)
+			 * - data (30 MiB)
+			 * - fota (99 MiB)
+			 */
+			partition@5fc0000 {
+				label = "rootfs";
+				reg = <0x5fc0000 0x9a00000>;
+			};
+		};
+	};
+};
+
+&qusb_phy_0 {
+	status = "okay";
+};
+
+&ssphy_0 {
+	status = "okay";
+};
+
+&usb_0 {
+	pinctrl-0 = <&usb_pwr_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&mdio {
+	status = "okay";
+
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+
+	qca8081_24: ethernet-phy@24 {
+		compatible = "ethernet-phy-id004d.d101";
+		reg = <24>;
+		reset-gpios = <&tlmm 25 GPIO_ACTIVE_LOW>;
+	};
+
+	qca8081_28: ethernet-phy@28 {
+		compatible = "ethernet-phy-id004d.d101";
+		reg = <28>;
+		reset-gpios = <&tlmm 44 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&switch {
+	status = "okay";
+
+	switch_lan_bmp = <ESS_PORT5>; /* lan port bitmap */
+	switch_wan_bmp = <ESS_PORT6>; /* wan port bitmap */
+	switch_mac_mode1 = <MAC_MODE_SGMII_CHANNEL0>; /* mac mode for uniphy instance1*/
+	switch_mac_mode2 = <MAC_MODE_SGMII_CHANNEL0>; /* mac mode for uniphy instance2*/
```

Comment:
Why not `MAC_MODE_SGMII_PLUS` as both of these are SGMII+ PHY-s?

Reply:
This is written based on the device tree of stock firmware. Edgecore EAP102
and Yuncore AX880 are also written like this, I'm not sure if it's correct.

Reply:
I know they are written like that, its probably copy/paste from old QSDK version.

Please try `MAC_MODE_SGMII_PLUS` and if it works(It should) use that.

Reply:
After a short period of testing, `MAC_MODE_SGMII_PLUS` can work properly.